aten/src/ATen/native/sparse/cuda/SparseCUDATensor.cu

```diff
@@ -152,4 +198,91 @@ SparseTensor coalesce_sparse_cuda(const SparseTensor& self) {
   return dst;
 }
 
+Tensor sparse_mask_helper_cuda(
+    const SparseTensor& t,
+    const Tensor& mask_indices) {
+  /*
+    This is a helper function which filter values from `t._values()` using the
+    `mask_indices`. This CUDA implementation uses `thrust::lower_bound`
+    operation to find the intersection of the `mask_indices` and the
+    `t._indices()` to then filter the values.
+
+    Inputs:
+      `t`             - tensor input
+      `mask_indices`  - mask indices tensor
+  */
+  int64_t r_nnz = mask_indices.size(1);
+  auto t_values = t._values().contiguous();
+  auto full_size = t.sizes();
+  auto vsize = t_values.sizes().vec();
+  vsize[0] = r_nnz;
+  Tensor r_values = at::zeros({vsize}, t_values.options());
+
+  auto t_indices = t._indices().contiguous();
+  auto t_nnz = t._nnz();
+
+  int curDevice = -1;
+  cudaGetDevice(&curDevice);
```

Comment:
I think the default `getCurrentCUDAStream()` will give the stream on the current device?

Reply:
yes you are right